parachain/pallets/ethereum-beacon-client/src/types.rs

```diff
@@ -2,12 +2,12 @@ pub use crate::config::{
 	SLOTS_PER_HISTORICAL_ROOT, SYNC_COMMITTEE_BITS_SIZE as SC_BITS_SIZE,
 	SYNC_COMMITTEE_SIZE as SC_SIZE,
 };
-use frame_support::storage::types::OptionQuery;
+use frame_support::{storage::types::OptionQuery, traits::ConstU32};
 use snowbridge_core::RingBufferMapImpl;
 
 // Specialize types based on configured sync committee size
 pub type SyncCommittee = primitives::SyncCommittee<SC_SIZE>;
-pub type SyncCommitteePrepared = primitives::SyncCommitteePrepared<SC_SIZE>;
+pub type SyncCommitteePrepared = primitives::SyncCommitteePrepared<ConstU32<{ SC_SIZE as u32 }>>;
```

Comment:
yeah, I am also curious about this change. Since the sync committee size is known at compile time, we should prefer to use a fixed-size array instead of a vec.

Reply:
Just fail when provisiong initial checkpoint for goerli and I'm open if there is an easy way to tackle it.

Reply:
Ah, ok, I understand. Did not see your previous comment to Clara.